doc/modules/clustering.rst

```diff
@@ -1300,3 +1300,244 @@ Drawbacks
   concepts of clusters, such as density based clusters like those obtained
   through DBSCAN.
 
+
+Select number of clusters
+===============================
+
+.. figure:: ../auto_examples/cluster/images/plot_chosen_nb_cluster_comparaison.png
+   :target: ../auto_examples/cluster/plot_chosen_nb_cluster_comparaison.html
+   :align: center
+   :scale: 50
+
+   A comparison of algorithms to select the number of clusters in scikit-learn. The clustering algorithm used is spectral clustering
+
+.. currentmodule:: sklearn.metrics.cluster
+
+Many algorithms require to select the wanted number of clusters. If one does not know how many clusters he wants, there exists algorithm to find the most relevant number of clusters for its data, given the data and the clustering algorithm used.
+
+.. _unsupervised.silhouette_coefficient:
+
+Silhouette Coefficient
+----------------------
+
+Presentation and usage
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. _stability:
+
+Stability
+---------
+
+Presentation
+~~~~~~~~~~~~
+
+A number of clusters is relevant if the clustering algorithm finds similar
+results with small perturbations of the data. In this implementation, we use
+the clustering algorithm on 2 large overlapping subset of the data. If the
+number of clusters is relevant, data is both subset should be clustered in
+a similar way. Let :math:`E` be the initial data, :math:`E_1` and :math:`E_2`
+be subsets of :math:`E`, each containing :math:`80\%` of :math:`E`.
+Let :math:`C_1` and :math:`C_2` be clusters obtained on datasets :math:`E_1`
+and :math:`E_2` Let :math:`A_i` be the adjacency matrix:
+
+.. math:: \left(A_i\right)_{ij} = \left\{
+	      \begin{tabular}
+	      1 & \mbox{if i and j belong to the same cluster} \\ 
+	      0 & \mbox{otherwise}
+	      \end{tabular}
+	  \right. 
+
+On points belonging to :math:`E_1` and :math:`E_2`, the adjacency matrix
+should stay the same. We compute the similarity between these two clustering
+
+.. math:: s(C_1, C_2) = \frac{
+   \sum_{(x_i, x_j) \in E_1 \cap E_2}\left(A_1\right)_{ij}\left(A_2\right)_{ij}}
+   {\sqrt{\sum_{(x_i, x_j) \in E_1 \cap E_2}\left(\left(A_1\right)_{ij}\right)^2}
+   \sqrt{\sum_{(x_i, x_j) \in E_1 \cap E_2}\left(\left(A_2\right)_{ij}\right)^2}}
```

Comment:
Maybe it's not so obscure, I just don't see it around: https://en.wikipedia.org/wiki/Fowlkes%E2%80%93Mallows_index